docs/sequenceDiagrams/seq-cep-10.4-action-flow.plantuml

```diff
@@ -0,0 +1,95 @@
+@startuml
+' declate title
+title 10.4. Notification Action Flow
+autonumber
+
+' declare actors
+collections "Notification-Topic" as topicNotify
+boundary "Central Event Processor" as CEP
+control "Action Agent" as AA
+database "CEP Store" as DB
+
+box "Central Services" #LightYellow
+participant topicNotify
+end box
+box "Central Event Processing Service" #LightGreen
+participant CEP
+participant AA
+participant DB
+end box
+
+' start flow
+Group Action Flow
+    activate CEP
+    CEP -> AA: Recieves an action event
+    activate AA
+    AA -> AA: Evaluate action event
+
+    alt **Action Object** exist
+        AA -> DB: Request **Event Object**  details
+        activate DB
+        deactivate DB
+        hnote over DB #LightYellow
+            event
+        end hnote
+        note right of AA #LightGrey
+            **Event Object** details;
+                Action details,
+                Hub notification details,
+                DFSP notification details.
+        end note
+        AA -> AA: Prepare notification details
+        AA -> DB: Request **Action Object** by same **Event Object** details
+        activate DB
+        deactivate DB
+        hnote over DB #LightYellow
+            action
+        end hnote
+
+        alt No previous **Action Object** with same event detail
+            AA -> DB: Store **Action Object**
+            activate DB
+            hnote over DB #LightYellow
+                action
+            end hnote
+            AA <-- DB: Return result **actionId**
+            deactivate DB
+            AA -> AA: Schedule reset repetition for **actionId** as per rules
+            ref over AA :  Notification Scheduler Flow {[[https://github.com/mojaloop/central-event-processor/tree/master/docs/sequenceDiagrams/seq-cep-10.5-scheduler-flow.svg 10.5]]} \n
+            note left of AA #yellow
+                Notification topic:
```

Comment:
If message is preserved, "Notification topic:" line may be deleted as the message is sent there.

Reply:
Line dropped.